docs/api.md

```diff
@@ -644,6 +644,33 @@ most.from([1, 2])
 	.observe(console.log.bind(console));
 ```
 
+### flatMapEnd
+
+####`stream.flatMapEnd(f) -> Stream`
+####`most.flatMapEnd(f, stream) -> Stream`
+
+Map the end event to a new stream, and begin emitting its values.  Note that `f` *must* return a stream.
+
+`function f(x) -> Stream`
+
+```
+stream:            -a-b-c-d-e-f->
+stream.take(4):         -a-b-c-d|
+f(end): 				-1-2-3-4-5->
+stream.flatMapEnd(f): -a-b-c-d-1-2-3-4-5>
+```
```

Comment:
Hmmm, after looking at the rendered markdown, I wonder if it's more clear like this:

```js
stream:               -a-b-c-d-e-f->
stream.take(4):       -a-b-c-d|
f(end):                       -1-2-3-4-5->
stream.flatMapEnd(f): -a-b-c-d-1-2-3-4-5->
```

What do you think?  It'd be nice to find a way to represent the end signal value in these diagrams.  A while back, someone had proposed something like the following:

```js
stream:               -a-b-c-d-e-f->
stream.take(4):       -a-b-c-d|x
f(x):                          1-2-3-4-5->
stream.flatMapEnd(f): -a-b-c-d-1-2-3-4-5->
```

I don't think we need to tackle that in this PR (since it has far-reaching documentation implications!), but just wanted to see what you think.